src/observers/slack_resource.cpp

```diff
@@ -23,39 +24,59 @@ Try<Nothing> SlackResourceObserver::consume(const ResourceUsage& usage) {
 
       auto previousSample = this->previousSamples->find(executor);
       if (previousSample != this->previousSamples->end()) {
-        Result<double_t> slackResource = CalculateCpuSlack(
-            (*previousSample), executor);
-        if (slackResource.isSome()) {
-            LOG(INFO) << NAME << "Estimated revocable resources for "
-                      << executor.executor_info().name() << " FrameworkID("
-                      << executor.executor_info().framework_id() << "): "
-                      << slackResource.get();
-            cpuSlack += slackResource.get();
-        } else if (slackResource.isError()) {
-          LOG(ERROR) << slackResource.error();
+        Result<double_t> executorCpuSlack = CalculateCpuSlack(
+            *previousSample, executor);
+        Try<double_t> executorCpuUsage = CountCpuUsage(
```

Comment:
Why do you calculate _the same_ Cpu Usage twice? I'd rather calculate `CpuUsage` first and then pass the `executorCpuUsage` to `CalculateCpuSlack` function if you particulary need to "encapsulate" that.